storage/remote/write.go

```diff
@@ -203,6 +203,19 @@ func (rws *WriteStorage) Close() error {
 	return nil
 }
 
+// Queues returns all of the queues that have been configured
+func (rws *WriteStorage) Queues() []*QueueManager {
```

Comment:
I wonder if rather than returning the actual queues if we should just return stats instead with a `Stats()` method? Not a strong feeling either way, but I am not sure if we need to get the full queues.

Reply:
> Awesome work so far!
> 
> > A few things I'm a bit unsure on is how we should best extract the pending samples, and last send, last send duration and last error status for each shard. My current thought is introducing a map to the shards of shard id to some kind of ShardStatus struct that contains each of these values and then to update this "status" struct from each shards goroutine.
> 
> That seems reasonable to me, an alternative would be a new `shard` struct that would then have a `run` method on it in place of `runShard`. The nice part of that would be each shard would then be independent of each other and not need to know about a shared state map. `shards` would then have to have a `[]*shard`, but that makes sense from a dependency perspective anyway.

I think that alternative is definitely what I would prefer potentially, I was just cautious of making significant changes to the working code. That being said, I think I will go with your suggestion, since it seems cleaner imho.

Reply:
Not too sure on how I feel on this one yet. Probably will build out the rest and then see if my opinion changes on this. It definitely feels odd to return the entire queues as you've said.